### PR TITLE
system/flatbuffers: fix patch flow of flatbuffers

### PR DESCRIPTION
**Pull request by anchao**
## Summary

system/flatbuffers: fix patch flow of flatbuffers

```
Applying patch to flatbuffers
patching file '<'
Hunk #1 FAILED at 1.
1 out of 1 hunk FAILED -- saving rejects to file '<.rej'
patching file '<'
Hunk #1 FAILED at 39.
1 out of 1 hunk FAILED -- saving rejects to file '<.rej'
patching file '<'
Hunk #1 FAILED at 495.
1 out of 1 hunk FAILED -- saving rejects to file '<.rej'
patching file '<'
Hunk #1 FAILED at 23.
1 out of 1 hunk FAILED -- saving rejects to file '<.rej'
patch: **** Can't reopen file '<' : No such file or directory
-- Configuring done (12.8s)
-- Generating done (0.3s)
```

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

N/A

## Testing

sim/nsh and enable CONFIG_SYSTEM_FLATBUFFERS

### CODE REVIEW

File: math/ruy/CMakeLists.txt [hunk 1]
```diff
@@ -44,17 +44,21 @@ if(CONFIG_MATH_RUY)
     endif()
   endif()
 
+  # cmake-format: off
```
Comment (xiaoxiang781216) [inline math/ruy/CMakeLists.txt:47]: why add

File: math/ruy/CMakeLists.txt [hunk 2]
```diff
@@ -44,17 +44,21 @@ if(CONFIG_MATH_RUY)
...
+
   # ############################################################################
   # Include Directory
   # ############################################################################
 
-  set(INCDIR ${CMAKE_CURRENT_LIST_DIR}/ruy)
+  # set(INCDIR ${CMAKE_CURRENT_LIST_DIR}/ruy)
```
Comment (xiaoxiang781216) [inline math/ruy/CMakeLists.txt:53]: why comment out

File: math/ruy/CMakeLists.txt [hunk 3]
```diff
@@ -44,17 +44,21 @@ if(CONFIG_MATH_RUY)
...
 
   # ############################################################################
   # Library Configuration
   # ############################################################################
 
-  nuttx_add_library(ruy STATIC)
-  target_include_directories(ruy PUBLIC ${INCDIR})
+  # nuttx_add_library(ruy STATIC)
```
Comment (xiaoxiang781216) [inline math/ruy/CMakeLists.txt:59]: ditto

File: mlearning/tflite-micro/CMakeLists.txt [hunk 4]
```diff
@@ -142,14 +135,20 @@ if(CONFIG_TFLITEMICRO)
   # Include Directory
   # ############################################################################
 
+  # cmake-format: off
```
Comment (xiaoxiang781216) [inline mlearning/tflite-micro/CMakeLists.txt:138]: ditto

File: math/ruy/CMakeLists.txt (hunk 1 above)
Comment (anchao) [reply math/ruy/CMakeLists.txt:47]: `cmake-format` tool cannot recognize the comment format. You need to turn off format detection here.

File: mlearning/tflite-micro/CMakeLists.txt (hunk 4 above)
Comment (anchao) [reply mlearning/tflite-micro/CMakeLists.txt:138]: cmake-format cannot correctly format the following include path

Comment (xiaoxiang781216) [reply mlearning/tflite-micro/CMakeLists.txt:144]: but it.work before?

File: math/ruy/CMakeLists.txt (hunk 2 above)
Comment (xiaoxiang781216) [reply math/ruy/CMakeLists.txt:54]: but why not fix it?

Comment (anchao) [reply math/ruy/CMakeLists.txt:54]: Many libraries depend on ruy and only need header file support, but tflm does not need it. Why spend time compiling it?and also, this is a issue submitted by the original author, and I have no interest in support it anymore

File: math/ruy/CMakeLists.txt (hunk 1 above)
Comment (xiaoxiang781216) [reply math/ruy/CMakeLists.txt:47]: but why not update the comment to make cmake-format happy

File: math/ruy/CMakeLists.txt (hunk 2 above)
Comment (xiaoxiang781216) [reply math/ruy/CMakeLists.txt:54]: if so, just remove the code instead comment out the unworked code.

Comment (anchao) [reply math/ruy/CMakeLists.txt:54]: No, but the header file of ruy is working

File: math/ruy/CMakeLists.txt (hunk 1 above)
Comment (anchao) [reply math/ruy/CMakeLists.txt:47]: The following line is a comment line, perhaps I should remove the logic below without any reservation?

Comment (xiaoxiang781216) [reply math/ruy/CMakeLists.txt:47]: yes, it's better to remove the code which doesn't work as expect.

Comment (anchao) [reply math/ruy/CMakeLists.txt]: Done

**Comment by anchao** [comment]
@xiaoxiang781216 done, please review again, ci is broken now

**Review by cederom — APPROVED** [review]
Looks and builds fine now, thank you @xiaoxiang781216 for the remarks  :-) Thank you @anchao for the updates :-)

**Comment by anchao** [comment]
I have tested this commit locally and found no other problems. Let's merge this commit first to resolve the CI broken.

**Comment by lupyuen** [comment]
Sorry @anchao Are we missing the file `boards/sim/sim/tflm/CMakeLists.txt`?

Or should we remove `CMake,sim:tflm` from from `tools/ci/testlist/sim-03.dat`?

https://github.com/NuttX/nuttx/actions/runs/12104485209/job/33747993566#step:7:377
```text
Cmake in present: sim/tflm
Configuration/Tool: sim/tflm
Changes not staged for commit:
	modified:   boards/sim/sim/sim/configs/tflm/defconfig
```

**Comment by anchao** [comment]
> Sorry @anchao Are we missing something in `CMakeLists.txt`?
> 
> Or should we remove `CMake,sim:tflm` from `tools/ci/testlist/sim-03.dat`?
> 
> https://github.com/NuttX/nuttx/actions/runs/12104485209/job/33747993566#step:7:377
> 
> ```
> Cmake in present: sim/tflm
> Configuration/Tool: sim/tflm
> Changes not staged for commit:
> 	modified:   boards/sim/sim/sim/configs/tflm/defconfig
> ```

@lupyuen I'm so sorry, I missed a config that was not refreshed correctly. I solved this problem in PR https://github.com/apache/nuttx/pull/15007,  Thank you very much for your reminder.

One more thing - I've noticed that building TFLM takes a considerable amount of time. If this puts additional strain on the community servers, it might be worth considering adding sim/tflm to the blacklist in the CI pipeline.

**Pull request merged**